### PR TITLE
Improvements/refactoring matching desings

### PR DESCRIPTION
**Pull request by RamonGilabert**

### CODE REVIEW

File: Source/Extensions/AutolayoutManager.swift [hunk 1]
```diff
@@ -0,0 +1,130 @@
+import UIKit
+
+// MARK: - BottomContainer autolayout
+
+extension BottomContainerView {
+
+  func setupConstraints() {
+
+    for attribute: NSLayoutAttribute in [.CenterX, .CenterY] {
```
Comment (richardoti) [inline Source/Extensions/AutolayoutManager.swift]: Of course, I understand, why all the hassle with not adding dependencies, but after trying Masonry/Cartography default  AutoLayout syntax looks really weird.

Nice idea to keep it away from logic-related code.

File: Source/ImagePickerController.swift [hunk 2]
```diff
@@ -340,20 +312,7 @@ extension ImagePickerController: ImageGalleryPanGestureDelegate {
           }
       })
     } else if velocity.y > 100 || galleryView.frame.size.height - galleryView.topSeparator.frame.height < 100 {
-      UIView.animateWithDuration(0.2, animations: { [unowned self] in
```
Comment (richardoti) [inline Source/ImagePickerController.swift]: I like to keep everything concise, so you can understand `expected` behavior just by reading code. Then, you can go to implementation details of particular method and fix it there, without being distracted by all the surrounding if/else/etc.
Good job!

File: Source/ImageGallery/ImageGalleryView.swift [hunk 3]
```diff
@@ -47,21 +47,21 @@ public class ImageGalleryView: UIView {
     let view = UIView()
     view.setTranslatesAutoresizingMaskIntoConstraints(false)
     view.addGestureRecognizer(self.panGestureRecognizer)
-    view.backgroundColor = self.configuration.backgroundColor
+    view.backgroundColor = UIColor.blackColor().colorWithAlphaComponent(0.6)
```
Comment (zenangst) [inline Source/ImageGallery/ImageGalleryView.swift]: This is cool :sunglasses:

**Comment by zenangst** [comment]
@hyperoslo/myshop-ios This looks awesome! What do you guys think?

File: Source/ImageGallery/ImageGalleryView.swift [hunk 4]
```diff
@@ -17,12 +17,12 @@ public class ImageGalleryView: UIView {
...
   struct Dimensions {
     static let galleryHeight: CGFloat = 160
-    static let galleryBarHeight: CGFloat = 34
+    static let galleryBarHeight: CGFloat = 24
     static let indicatorWidth: CGFloat = 41
     static let indicatorHeight: CGFloat = 8
   }
 
-  lazy public var collectionView: UICollectionView = { [unowned self] in
+  lazy public var collectionView: UICollectionView = {
```
Comment (vadymmarkov) [inline Source/ImageGallery/ImageGalleryView.swift]: `[unowned self] in`? :smile:

Comment (RamonGilabert) [inline Source/ImageGallery/ImageGalleryView.swift]: Fuck! :smile:

**Comment by vadymmarkov** [comment]
@RamonGilabert Looks great!

**Comment by RamonGilabert** [comment]
:sake:

**Comment by zenangst** [comment]
:coffee:

File: Source/BottomView/ImageStack.swift [hunk 5]
```diff
@@ -3,9 +3,9 @@ import UIKit
 public class ImageStack {
 
   public struct Notifications {
-    public static let imageDidPush = "imageDidPush:"
```
Comment (richardoti) [inline Source/BottomView/ImageStack.swift]: Hmmm. This needs to be checked.

Comment (RamonGilabert) [inline Source/BottomView/ImageStack.swift]: It's a notification name, not a method. The name of the notification is imageDidPush, not imageDidPush:.

File: Source/ImagePickerController.swift [hunk 6]
```diff
@@ -258,102 +269,48 @@ extension ImagePickerController: ImageGalleryPanGestureDelegate {
...
-    CATransaction.setDisableActions(true)
-    cameraController.previewLayer?.frame.size.height = galleryView.frame.origin.y - topView.frame.height
-    CATransaction.commit()
     galleryView.noImagesLabel.center = galleryView.collectionView.center
-    if let targetIndexPath = targetIndexPath {
-      galleryView.collectionView.scrollToItemAtIndexPath(targetIndexPath, atScrollPosition: .CenteredHorizontally, animated: false)
-    }
   }
 
-  func panGestureDidEnd(translation: CGPoint, location: CGPoint, velocity: CGPoint) {
```
Comment (richardoti) [inline Source/ImagePickerController.swift]: :+1:

File: Source/BottomView/ImageStack.swift (hunk 5 above)
Comment (richardoti) [inline Source/BottomView/ImageStack.swift]: Yep, I did few tests to check that it works correctly. Seems that I've just copied selector.

**Pull request merged**